genie-web/src/main/java/com/netflix/genie/web/controllers/JobSearchAspect.java

```diff
@@ -0,0 +1,122 @@
+        } catch (final Throwable t) {
+            MetricsUtils.addFailureTagsWithException(tags, t);
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            log.debug("HealthEndpoint.invoke() completed in {} ns", turnaround);
+            registry
+                .timer(registry.createId("health-check.endpoint.timer", tags))
+                .record(turnaround, TimeUnit.NANOSECONDS);
+        }
+        return h;
+    }
+
+    @Around(
+        "execution("
+            + "  org.springframework.boot.actuate.health.Health "
+            + "  org.springframework.boot.actuate.health.HealthIndicator.health()"
+            + ")"
+    )
+    public Health healthIndicatorHealthMonitor(ProceedingJoinPoint joinPoint) throws Throwable {
+        final long start = System.nanoTime();
+        final Health h;
+        Throwable throwable = null;
+        try {
+            h = (Health) joinPoint.proceed(joinPoint.getArgs());
+        } catch (Throwable t) {
+            throwable = t;
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            recordHealthIndicatorTurnaround(turnaround, joinPoint, throwable);
+        }
+        return h;
+    }
+
+    @Around(
+        "execution("
+            + "  void " +
+            "    org.springframework.boot.actuate.health.AbstractHealthIndicator.doHealthCheck("
+            + "    org.springframework.boot.actuate.health.Health.Builder"
+            + "  )"
+            + ")"
+    )
+    public void AbstractHealthIndicatorDoHealthCheckMonitor(ProceedingJoinPoint joinPoint) throws Throwable {
+        final long start = System.nanoTime();
+        Throwable throwable = null;
+        try {
+            joinPoint.proceed(joinPoint.getArgs());
+        } catch (Throwable t) {
+            throwable = t;
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            recordHealthIndicatorTurnaround(turnaround, joinPoint, throwable);
+        }
+    }
+
+    private void recordHealthIndicatorTurnaround(
+        long turnaround,
+        ProceedingJoinPoint joinPoint,
+        @Nullable Throwable throwable
+    ) {
+        log.debug(
+            "{} completed in {} ns (exception: {})",
+            joinPoint.getTarget().getClass().getSimpleName(),
+            turnaround,
+            throwable != null ? throwable.getClass().getSimpleName() : "none"
+        );
+        final Map<String, String> tags;
+        if (throwable == null) {
+             tags = MetricsUtils.newSuccessTagsMap();
+        } else {
+            tags = MetricsUtils.newFailureTagsMapForException(throwable);
+        }
+        tags.put(HEALTH_INDICATOR_NAME, joinPoint.getTarget().getClass().getSimpleName());
+        registry
+            .timer(registry.createId("health-check.indicator.timer", tags))
```

Comment:
don't like this name. Haven't we followed `genie.{metricClass}.{name}.{type}` kind of naming syntax for all our other ones? I like at least having `genie` at the front to indicate that it's something we've created rather than something coming from Boot or other system

Reply:
Yes this was a placeholder, going to change to follow convention